src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -136,6 +142,67 @@ public function editAction(Request $request, int $productId): Response
         return $this->renderProductForm($productForm, $productId);
     }
 
+    /**
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))", message="You do not have permission to update this.")
+     *
+     * @param int $productId
+     *
+     * @return JsonResponse
+     */
+    public function getImagesAction(int $productId): JsonResponse
+    {
+        /** @var ProductImage[] $images */
+        $images = $this->getQueryBus()->handle(new GetProductImages($productId));
+
+        $data = [];
+        foreach ($images as $image) {
+            $data[] = [
+                'image_id' => $image->getImageId(),
+                'is_cover' => $image->isCover(),
+                'position' => $image->getPosition(),
+                'path' => _THEME_PROD_DIR_ . $image->getPath(),
+                'legends' => $image->getLocalizedLegends(),
+            ];
+        }
+
+        return new JsonResponse($data);
+    }
+
+    /**
+     * @AdminSecurity("is_granted(['update'], request.get('_legacy_controller'))", message="You do not have permission to update this.")
+     *
+     * @param Request $request
+     * @param int $productId
+     *
+     * @return JsonResponse
+     */
+    public function addImageAction(Request $request, int $productId): JsonResponse
+    {
+        $imageForm = $this->createForm(AddImageType::class);
+        $imageForm->handleRequest($request);
+
+        if (!$imageForm->isSubmitted() || $imageForm->isValid()) {
+            return new JsonResponse([
+                'error' => 'Invalid data.',
+                'form_errors' => $this->getFormErrorsForJS($imageForm),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $formData = $imageForm->getData();
+        $uploadedFile = $formData['file'];
+        try {
+            $command = new AddProductImageCommand($productId, $uploadedFile->getPathname());
+            /** @var ImageId $imageId */
+            $imageId = $this->getCommandBus()->handle($command);
+        } catch (Exception $e) {
+            return new JsonResponse([
+                'error' => $this->getErrorMessageForException($e, $this->getErrorMessages($e)),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        return new JsonResponse(['image_id' => $imageId->getValue()]);
```

Comment:
Shouldn't have to return the same array table as  above?

Reply:
Yes good idea, I added a comment for this I can't do it right now because the `GetProductImageCommand` is implemented in another PR https://github.com/PrestaShop/PrestaShop/pull/23661